gandi/resource_livedns_key.go

```diff
@@ -0,0 +1,173 @@
+package gandi
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceLiveDNSKey() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Domain name",
+			},
+			"flags": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "DNSSEC key flags",
+			},
+			"algorithm": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "DNSSEC algorithm type",
+			},
+			"algorithm_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "DNSSEC algorithm name",
+			},
+			"deleted": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Is the key deleted?",
+			},
+			"ds": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "DS record as RFC1035 line",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Current status of the key",
+			},
+			"public_key": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Public key",
+			},
+			"tag": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Tag",
+			},
+		},
+		CreateContext: resourceLiveDNSKeyCreate,
+		Delete:        resourceLiveDNSKeyDelete,
+		Read:          resourceLiveDNSKeyRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func resourceLiveDNSKeyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients).LiveDNS
+	domain := d.Get("domain").(string)
+
+	// The API does not return the key UUID. Not very convenient.
+	// We will assume we will use the last key.
+	last := ""
+	keys, err := client.GetDomainKeys(domain)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if len(keys) > 0 {
+		last = keys[len(keys)-1].UUID
+	}
+
+	_, err = client.SignDomain(domain)
```

Comment:
I think you should be able to get the resource UUID by using the Location response header such as in https://github.com/go-gandi/go-gandi/blob/8ada87e22651f84dc33b6365f0d1dcad5c483e45/simplehosting/simplehosting.go#L48

Reply:
Thanks! I'll do that!